enhancements/update/object-removal-manifest-annotation.md

```diff
@@ -0,0 +1,228 @@
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Currently CVO managed object removal is handled by jobs.
+Jobs have many immutable components, so they aren't a great match for CVO managed manifests.
+This enhancement replaces that approach with a new manifest annotation requesting the CVO delete the in-cluster object instead of creating/updating it.
+This will provide a more straightforward way for engineering to remove content.
+
+## Motivation
+
+There should be a straightforward way to remove CVO managed objects.
+
+### Goals
+
+* Developers will be able to remove any of the currently managed CVO objects by modifying an existing manifest and adding the new delete annotation.
+
+### Non-Goals
+
+* This enhancement does not add any new objects for management by CVO, therefore the new delete annotation only applies to the currently supported manifests and any that may be added in the future.
+* This enhancement neither adds nor requires any application specific delete logic within CVO.
+
+## Proposal
+
+When the following annotation appears in a CVO supported manifest and is set to "true" the associated object will be removed from the cluster by the CVO.
+Values other than `true` will result in a CVO failure.
+However this should never occur in a release since any value other than `true` will result in CVO CI failure.
+```
+apiVersion: apps/v1
+...
+metadata:
+...
+  annotations:
+    release.openshift.io/delete: "true"
+```
+The existing CVO ordering scheme defined [here](https://github.com/openshift/cluster-version-operator/blob/master/docs/dev/operators.md#what-is-the-order-that-resources-get-createdupdated-in) will also be used for object removal.
+This both minimizes CVO code changes and provides a simple method of deleting multiple objects by reversing the order in which they were created.
+It is the developer's responsibility to ensure proper deletion ordering and to ensure that all items originally created by an object are deleted when that object is deleted.
+For example, an operator may have to be modified, or a new operator created, to take explicit actions to remove external resources.
+The modified or new operator would then be removed in a subsequent update.
+
+### Delete Finalization
+
+Two delete finalization options are described below to help reach a consensus on the best approach.
+
+#### Finalization Option 1
+
+The first approach is to handle deletion requests similar to how CVO handles create/update requests.
+This is in a non-blocking manner whereby CVO issues the initial request to delete an object kicking off resource finalization and after which resource removal.
+CVO does not wait for actual resource removal but instead continues.
+As CVO manifest processing continues the object's `.metadata.deletionTimestamp`, set when the delete was originally requested, informs CVO that this object has already been processed for deletion.
```

Comment:
Will the CVO report that it is "done" before the resource has been finalized?  I like the idea of continuing through the payload, but I'd like to be able to look at the ClusterVersion.status and know whether I'm still waiting for deletion.  This seems comparable to waiting on clusteroperators, but not blocking on them.

Reply:
"done" would be after it's finalized and gone. Given a valid manifest delete request I see three possible "states" CVO would recognize, and report: delete initiated, ongoing if we process the manifest again and find it already has a deletion timestamp, and done.

Reply:
Added additional details regarding continued processing of delete manifest.